internal/middleware/middleware_test.go

```diff
@@ -0,0 +1,569 @@
+				req.AddCookie(&http.Cookie{
+					Name:  "token",
+					Value: "jwtToken",
+				})
+				return req, mockJwtSvc
+			},
+			validator: func(res *httptest.ResponseRecorder, hit *bool) {
+				if *hit != true {
+					t.Errorf("Want: %v, Got: %v", true, *hit)
+					return
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				result := model.Response{}
+				err := json.Unmarshal(by, &result)
+				if err != nil {
+					t.Log(err)
+					return
+				}
+				expected := &model.Response{
+					Status:  http.StatusOK,
+					Message: "passed",
+					Data:    "123",
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+		{
+			name: "Failure::ExtractUser:: empty token value",
+			setupFunc: func() (*http.Request, authentication.JWTService) {
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				mockJwtSvc := mock.NewMockJWTService(mockCtrl)
+				req.AddCookie(&http.Cookie{
+					Name:  "token",
+					Value: "",
+				})
+				return req, mockJwtSvc
+			},
+			validator: func(res *httptest.ResponseRecorder, hit *bool) {
+				if *hit != false {
+					t.Errorf("Want: %v, Got: %v", false, hit)
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				result := model.Response{}
+				err := json.Unmarshal(by, &result)
+				if err != nil {
+					t.Log(err)
+					return
+				}
+				expected := &model.Response{
+					Status:  http.StatusUnauthorized,
+					Message: codes.GetErr(codes.ErrUnauthorized),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+		{
+			name: "Failure::ExtractUser:: no cookie found",
+			setupFunc: func() (*http.Request, authentication.JWTService) {
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				mockJwtSvc := mock.NewMockJWTService(mockCtrl)
+				return req, mockJwtSvc
+			},
+			validator: func(res *httptest.ResponseRecorder, hit *bool) {
+				if *hit != false {
+					t.Errorf("Want: %v, Got: %v", false, hit)
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				result := model.Response{}
+				err := json.Unmarshal(by, &result)
+				if err != nil {
+					t.Log(err)
+					t.Fail()
+					return
+				}
+				expected := &model.Response{
+					Status:  http.StatusUnauthorized,
+					Message: codes.GetErr(codes.ErrUnauthorized),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+		{
+			name: "Failure::ExtractUser:: compared literals not same",
+			setupFunc: func() (*http.Request, authentication.JWTService) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				req.AddCookie(&http.Cookie{
+					Name:  "token",
+					Value: " jwtToken",
+				})
+				mockJwtSvc := mock.NewMockJWTService(mockCtrl)
+				err := errors.New(" err ")
+				mockJwtSvc.EXPECT().ValidateToken(gomock.Any()).Return(nil, err)
```

Comment:
Why is `gomock.Any()` used for this? Looks like this is used in many mocks.

Reply:
done